components/script/dom/globalscope.rs

```diff
@@ -3193,32 +3194,54 @@ impl GlobalScope {
 
         // <https://w3c.github.io/gamepad/#dfn-update-gamepad-state>
         self.gamepad_task_source()
-            .queue(
+            .queue_with_canceller(
                 task!(update_gamepad_state: move || {
                     let global = this.root();
                     if let Some(window) = global.downcast::<Window>() {
-                        let gamepad_list = window.Navigator().GetGamepads();
-                        if let Some(gamepad) = gamepad_list.IndexedGetter(index as u32) {
+                        let gamepad_list = window.Navigator().gamepads();
+                        if let Some(gamepad) = gamepad_list.Item(index as u32) {
                             let current_time = global.performance().Now();
                             gamepad.update_timestamp(*current_time);
 
+                            let mut contains_user_gesture = false;
```

Comment:
let's put this in a separate function taking the update type as argument and with a doclink to https://w3c.github.io/gamepad/#dfn-gamepad-user-gesture